### PR TITLE
MM-57375: Update to latest minio image

### PR DESCRIPTION
**Pull request by agnivade**
Updated necessary code to handle latest Minio version

https://mattermost.atlassian.net/browse/MM-57375

```release-note
NONE
```

### CODE REVIEW

File: server/platform/shared/filestore/s3store.go [hunk 1]
```diff
@@ -56,6 +56,7 @@ type S3FileBackendNoBucketError struct{}
 const (
 	// This is not exported by minio. See: https://github.com/minio/minio-go/issues/1339
 	bucketNotFound = "NoSuchBucket"
+	invalidBucket  = "InvalidBucketName"
```
Comment (agnivade) [inline server/platform/shared/filestore/s3store.go:59]: With the new version, there is a distinction between no bucket, and invalid bucket name. So we have to handle that as well.

**Review by streamer45 — APPROVED** [review]
Nice! I was fearing much more involved changes :sweat_smile:

File: server/platform/shared/filestore/s3store.go (hunk 1 above)
Comment (cpoile) [reply server/platform/shared/filestore/s3store.go:59]: That's a good distinction

**Review by cpoile — APPROVED** [review]

**Comment by agnivade** [comment]
Tests all passed. Since all unit tests and e2e tests have passed, going to merge this. Don't think any further manual testing is necessary.

File: server/build/docker-compose.common.yml [hunk 2]
```diff
@@ -59,14 +59,14 @@ services:
       timeout: 10s
       retries: 3
   minio:
-    image: "minio/minio:RELEASE.2019-10-11T00-38-09Z"
+    image: "minio/minio:RELEASE.2024-06-22T05-26-45Z"
```
Comment (tnir) [inline server/build/docker-compose.common.yml:62]: 🎉

**Review by tnir — COMMENTED** [review]
🎉

**Pull request merged**